pkg/dataobj/explorer/ui/src/components/file-metadata/FileMetadata.tsx

```diff
@@ -0,0 +1,445 @@
+import React, { useState } from "react";
+import { formatBytes } from "../../utils/format";
+import { DateWithHover } from "../common/DateWithHover";
+import { Link } from "react-router-dom";
+import { useBasename } from "../../contexts/BasenameContext";
+import { CompressionRatio } from "./CompressionRatio";
+
+interface PageInfo {
+  compressed_size: number;
+  uncompressed_size: number;
+  rows_count: number;
+  values_count: number;
+  encoding: string;
+  data_offset: number;
+  data_size: number;
+}
+
+interface ColumnInfo {
+  name?: string;
+  type: string;
+  value_type: string;
+  rows_count: number;
+  compression: string;
+  uncompressed_size: number;
+  compressed_size: number;
+  metadata_offset: number;
+  metadata_size: number;
+  values_count: number;
+  pages: PageInfo[];
+}
+
+interface SectionMetadata {
+  type: string;
+  totalCompressedSize: number;
+  totalUncompressedSize: number;
+  columnCount: number;
+  columns: ColumnInfo[];
+}
+
+interface FileMetadataProps {
+  metadata: {
+    sections: SectionMetadata[];
+    error?: string;
+    lastModified: string;
+  };
+  filename: string;
+  className?: string;
+}
```

Comment:
It's hard to tell in GitHub since the PR is big, but aren't these already defined in types/metadata.ts?

Reply:
Yes good catch that is leftover from refactoring